bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,179 @@ def get_parsed_info(tmp_path, file_contents):
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct AStruct"
+
+
+def test_constructor(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+
+    assert constructor["kind"] == "CONSTRUCTOR"
+    assert constructor["access_specifier"] == "PUBLIC"
+    assert constructor["name"] == "AStruct"
+
+
+def test_parm_decl(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct(int aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+    parm_decl = constructor["members"][0]
+
+    assert parm_decl["kind"] == "PARM_DECL"
+    assert parm_decl["element_type"] == "Int"
+    assert parm_decl["name"] == "aFunctionArgument"
+
+
+def test_unexposed_expr(tmp_path):
+    file_contents = """
+    struct AStruct {
+        int aClassMember;
+        AStruct(int aFunctionArgument) : aClassMember(aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][1]
+    unexposed_expr = constructor["members"][2]
+
+    assert unexposed_expr["kind"] == "UNEXPOSED_EXPR"
+    assert unexposed_expr["name"] == "aFunctionArgument"
+
+
+# @TODO: Not sure how to reproduce. Maybe later.
+# def test_member_ref_expr(tmp_path):
+
+
+def test_decl_ref_expr(tmp_path):
+    file_contents = """
+    struct AStruct {
+        int firstMember, secondMember;
+        AStruct(int firstFunctionArgument, int secondFunctionArgument)
+        : firstMember(secondFunctionArgument), secondMember(firstFunctionArgument)
+        {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][2]
+    unexposed_expr_1 = constructor["members"][3]
+    unexposed_expr_2 = constructor["members"][5]
+    decl_ref_expr_1 = unexposed_expr_1["members"][0]
+    decl_ref_expr_2 = unexposed_expr_2["members"][0]
+
+    assert decl_ref_expr_1["kind"] == "DECL_REF_EXPR"
+    assert decl_ref_expr_2["kind"] == "DECL_REF_EXPR"
+    assert decl_ref_expr_1["name"] == "secondFunctionArgument"
+    assert decl_ref_expr_2["name"] == "firstFunctionArgument"
```

Comment:
https://github.com/divmadan/pcl/pull/6#discussion_r459610274

Reply:
Such cases makes a good case to use @kunaltyagi's for-loop approach in https://github.com/divmadan/pcl/pull/6#discussion_r456865271

```py
for member in parsed_info['members']:
  # do_stuff
```

Reply:
Seems clear to me, if you think it'd be clearer I'm all ears. It actually seems to parse in order from both this test and [the last one](https://github.com/divmadan/pcl/blob/9b8e504dd1fd1a36554cd818d580d29f4e7c865e/bindings/python/tests/test_parse.py#L190-L210).

Reply:
Yep, it does parses in order. The case to be made was about future robustness, if that will be an issue.
I'm fine either way.

Reply:
Sure, sure, forgot about that, it's correct. Can you show me how you'd write it?

Reply:
I got another approach. Consider this test: https://github.com/divmadan/pcl/blob/d1ebed1911c986638930ff34e724aa7c52dfe3e4/bindings/python/tests/test_parse.py#L123-L137

Suggestion:

```py
# add a function to get all dict items in the form of a list
def get_members(item, member_list):
    member_list.append(item)
    for member in item["members"]:
        get_members(member, member_list)


"""
List items:
{'depth': 0, 'line': 0, 'column': 0, 'kind': 'TRANSLATION_UNIT', 'name': '/tmp/pytest-of-divyanshu/pytest-91/test_parm_decl0/file.hpp', 'members': [{'depth': 1, 'line': 2, 'column': 12, 'kind': 'STRUCT_DECL', 'name': 'AStruct', 'element_type': 'Record', 'members': [{'depth': 2, 'line': 3, 'column': 9, 'kind': 'CONSTRUCTOR', 'name': 'AStruct', 'element_type': 'FunctionProto', 'access_specifier': 'PUBLIC', 'result_type': 'void', 'members': [{'depth': 3, 'line': 3, 'column': 21, 'kind': 'PARM_DECL', 'name': 'aFunctionParameter', 'element_type': 'Int', 'members': []}, {'depth': 3, 'line': 3, 'column': 41, 'kind': 'COMPOUND_STMT', 'name': '', 'members': []}]}]}]}
{'depth': 1, 'line': 2, 'column': 12, 'kind': 'STRUCT_DECL', 'name': 'AStruct', 'element_type': 'Record', 'members': [{'depth': 2, 'line': 3, 'column': 9, 'kind': 'CONSTRUCTOR', 'name': 'AStruct', 'element_type': 'FunctionProto', 'access_specifier': 'PUBLIC', 'result_type': 'void', 'members': [{'depth': 3, 'line': 3, 'column': 21, 'kind': 'PARM_DECL', 'name': 'aFunctionParameter', 'element_type': 'Int', 'members': []}, {'depth': 3, 'line': 3, 'column': 41, 'kind': 'COMPOUND_STMT', 'name': '', 'members': []}]}]}
{'depth': 2, 'line': 3, 'column': 9, 'kind': 'CONSTRUCTOR', 'name': 'AStruct', 'element_type': 'FunctionProto', 'access_specifier': 'PUBLIC', 'result_type': 'void', 'members': [{'depth': 3, 'line': 3, 'column': 21, 'kind': 'PARM_DECL', 'name': 'aFunctionParameter', 'element_type': 'Int', 'members': []}, {'depth': 3, 'line': 3, 'column': 41, 'kind': 'COMPOUND_STMT', 'name': '', 'members': []}]}
{'depth': 3, 'line': 3, 'column': 21, 'kind': 'PARM_DECL', 'name': 'aFunctionParameter', 'element_type': 'Int', 'members': []}
{'depth': 3, 'line': 3, 'column': 41, 'kind': 'COMPOUND_STMT', 'name': '', 'members': []}
"""

# updated function
def test_parm_decl(tmp_path):
    file_contents = """
    struct AStruct {
        AStruct(int aFunctionParameter) {}
    };
    """
    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)

    member_list = []
    get_members(parsed_info, member_list)
    
    flag = False
    for item in member_list:
        if item["kind"] == "PARM_DECL":
            assert item["element_type"] == "Int"
            assert item["name"] == "aFunctionParameter"
            flag = True
    assert flag
```

This incorporates both the `container` approach (@aponza's initial suggestion, `assert desired in str(parsed_info)`) and the `use a loop, don't make indexing assumptions` approach (@kunaltyagi's suggestion).